app/controllers/project_approvals_controller.rb

```diff
@@ -0,0 +1,21 @@
+# Allows a project to be approved
+class ProjectApprovalsController < ModerationController
+  expose(:project_approval, scope: -> { pending_project.project_status_changes },
+                            model: :project_status_change)
+  expose(:pending_project, scope: -> { Project.pending }, model: :project)
+
+  def new; end
+
+  def create
+    if project_approval.save
```

Comment:
Nonblocking: the lack of a transaction around the save and approve is making me twitchy.

Reply:
Good point! How do you feel about wrapping all controller actions in a transaction, or is that :too_much:? (Not right this instant, but it'd be one less thing to wrry about)